src/gstreamer/gstkvssink.cpp

```diff
@@ -663,6 +659,9 @@ gst_kvs_sink_finalize(GObject *object) {
     if (kvssink->stream_tags) {
         gst_structure_free (kvssink->stream_tags);
     }
+    if (data->kinesis_video_producer) {
```

Comment:
I see what was happening here. We were tearing down the stream only and leaving the producer object. Tearing down the SP will destruct the producer object and in turn will free all of the associated streams.